markets/bfp-market/test/integration/modules/PerpAccountModule.splitAccount.test.ts

```diff
@@ -0,0 +1,260 @@
+        toTrader.accountId,
+        marketId,
+        bn(0.1)
+      ),
+      `OrderFound()`,
+      PerpMarketProxy
+    );
+  });
+
+  it('should revert when toAccount have collateral', async () => {
+    const { PerpMarketProxy } = systems();
+
+    // Create two trader objects with different accountIds but same signer.
+    const fromTrader = genOneOf(traders());
+    const toTraderAccountId = 42069;
+    const toTrader = {
+      signer: fromTrader.signer,
+      accountId: toTraderAccountId,
+    };
+    await PerpMarketProxy.connect(toTrader.signer)['createAccount(uint128)'](toTraderAccountId);
+    const { marketId } = await depositMargin(bs, genTrader(bs, { desiredTrader: toTrader }));
+
+    await assertRevert(
+      PerpMarketProxy.connect(fromTrader.signer).splitAccount(
+        fromTrader.accountId,
+        toTrader.accountId,
+        marketId,
+        bn(0.1)
+      ),
+      `CollateralFound()`,
+      PerpMarketProxy
+    );
+  });
+
+  it('should revert when toAccount have a position', async () => {
+    const { PerpMarketProxy } = systems();
+
+    // Create two trader objects with different accountIds but same signer.
+    const fromTrader = genOneOf(traders());
+    const toTraderAccountId = 42069;
+    const toTrader = {
+      signer: fromTrader.signer,
+      accountId: toTraderAccountId,
+    };
+    await PerpMarketProxy.connect(toTrader.signer)['createAccount(uint128)'](toTraderAccountId);
+    const { marketId, market, collateral, collateralDepositAmount } = await depositMargin(
+      bs,
+      genTrader(bs, { desiredTrader: toTrader })
+    );
+    await commitAndSettle(
+      bs,
+      marketId,
+      toTrader,
+      genOrder(bs, market, collateral, collateralDepositAmount)
+    );
+
+    await assertRevert(
+      PerpMarketProxy.connect(fromTrader.signer).splitAccount(
+        fromTrader.accountId,
+        toTrader.accountId,
+        marketId,
+        bn(0.1)
+      ),
+      `PositionFound("${toTrader.accountId}", "${marketId}")`,
+      PerpMarketProxy
+    );
+  });
+
+  it('should revert when fromAccount is flagged');
+
+  it('should revert when fromAccount is liquidatable');
+
+  it('should split account', async () => {
+    const { PerpMarketProxy } = systems();
+
+    // Create two trader objects with different accountIds but same signer.
+    const fromTrader = genOneOf(traders());
+    const toTraderAccountId = 42069;
+    const toTrader = {
+      signer: fromTrader.signer,
+      accountId: toTraderAccountId,
+    };
+    await PerpMarketProxy.connect(toTrader.signer)['createAccount(uint128)'](toTraderAccountId);
+    const { marketId, market, collateral, collateralDepositAmount } = await depositMargin(
+      bs,
+      genTrader(bs, { desiredTrader: fromTrader })
+    );
+    await commitAndSettle(
+      bs,
+      marketId,
+      fromTrader,
+      genOrder(bs, market, collateral, collateralDepositAmount)
+    );
+
+    const proportion = bn(genNumber(0.1, 1));
```

Comment:
I would have an explicit test for the `1` case

Reply:
Done!